### ISSUE TITLE
Mutation and CNA table do not update after selection, respectively

### ISSUE THREAD

**Issue by awernibi** [cBioPortal/cbioportal#1269]
When I select a mutation in the study view, all other visualizations (pie charts, barplots, the cna table, kaplan meier, mutation count) update according to my selection.
Only the mutation table stays the same.

![image](https://cloud.githubusercontent.com/assets/13519768/15463780/c59a25a6-20cb-11e6-9abc-e42250e97932.png)

When I select a copy number aberation afterwards, the mutation table changes properly
![image](https://cloud.githubusercontent.com/assets/13519768/15463801/e971ffd0-20cb-11e6-9030-bd2d144d5ce0.png)

When I de-select the CNA again, both tables show the correct data:
![image](https://cloud.githubusercontent.com/assets/13519768/15463818/0fa04220-20cc-11e6-81e0-81cc0559c5b4.png)

In short, dependent on the sequence of selections, some tables show different data. (compare my image 1 and 3. They have exactly one mutation selected, but the percentages and numbers are different in the left panel)
This is also true when I start with a CNA and select a mutation afterwards and de-select the mutation again.

(don't tell me this is a feature ;)

**Comment by pieterlukasse** [comment]
@zhx828 : is this maybe similar to #176?

**Comment by zhx828** [comment]
@awernibi @pieterlukasse Unfortunately, it is a feature. And this is one of most debatable feature too. 
In the first screenshot, you do filtering among 200 profiled samples in Mutated Genes(MG) table, the MG shouldn't be updated because of the denominator hasn't been changed yet. After selecting PRSS8 in CNA table, the selected cases has been narrowed down to 2 which change the dominator of MG to 2, but since you do filtering in CNA table, it still has 52 profiled samples. In the third screenshot, after deselecting PRSS8, since CNA has 52 profiled samples(not 200) which is the current denominator, the rest of charts should be changed based on this number.

**Comment by awernibi** [comment]
The behavior of this small visualizations is quite inconsistent. 

The _mutation count vs. CNA_ scatter plot always shows all samples in blue and sometimes in red (regardless where the filtering comes from either (1) by clicking on another visualization like the mutation table or (2) by selecting samples on the scatterplot itself).

The Kaplan-Meier also shows always all samples (but splits the samples into a red and blue fraction. It does not support the selection of samples, which means that Kaplan maier is not directly involved into the selection process)

All other visualisation (pie charts, barplots, and tables) show different content, if they are involved into the selection process (e.g. by clicking on a mutated gene) or if the selection is triggered somewhere else.

To make it more intuitive is currently quite difficult:
* Survival (Kaplan Meier) curves makes only sense, if you show all samples (the selected and de-selected ones)
* The _mutation count vs. CNA_ scatter plot could show all samples (currently) or filter to the selected samples.
* in case of all other visualizations (pie charts, barplots, and tables) you could see the overlaps (image 3 above) or the original alteration frequency (image 1 above). 

This requires some training for new users, since it is not very intuitive.

**Comment by pieterlukasse** [comment]
@zhx828 I think the behavior is not correct if you consider figure 3 in more detail. The denominator (which is shown on top of the table, labelled as "X profiled samples") should be the total number of samples filtered by all filter criteria, except the filter criterion of the component itself. This means that the denominator for Mutated Genes in figure 3 should be 200. In the same way, we don't change a pie chart to a solid circle of 1 color when we select a section of it. See example below:

Step1: select Race category:
![image](https://cloud.githubusercontent.com/assets/2900303/15501262/de40c344-21ac-11e6-8a17-cf572b51c316.png)

Step2: select Overall Survival Status:
![image](https://cloud.githubusercontent.com/assets/2900303/15501277/ec3360ec-21ac-11e6-9793-0d6eb2d391ca.png)

Step3: deselect Overall Survival Status: this brings back the original pie chart (and thus the original denominator). The way the table *now* works would be similar to this picture being a solid orange circle with 183 samples, which would be wrong.
![image](https://cloud.githubusercontent.com/assets/2900303/15501323/28f6160a-21ad-11e6-8153-072bc0680e56.png)

@schultzn, @jjgao : what do you think? Have you heard similar feedback on this part?

**Comment by schultzn** [comment]
I am having trouble following this conversation. Are you concerned about the general logic, or are you simply saying there is a bug (see above)?

If the former, here is my explanation for why we have built I know this is tricky, and the current functionality, while not flexible enough to support any analysis a user might want to do, it is at least coherent (I think). Here is how I see it:

What the study view provides is the ability to select a set of samples for analysis. A user can select one or multiple entries of a given clinical attribute (or list of altered genes). Within each attribute, the selection follows the OR language, e.g. a user can select samples that are Stage III or Stage IV. Similarly, a user can select samples that have either a TP53 or a KRAS mutation. 

When combining selections from different attributes, the selection changes to an AND. So a user selects all samples that are stage III or IV and have a TP53 or KRAS mutation. 

The scatter plot shows the samples that match the selection and shows them in red. The Kaplan Meier plot shows the selected samples in the red curve and compares them to all other samples (not selected, blue).

That is indeed the functionality we had spec'd out. A key missing functionality is that a user cannot see the frequency of mutations in other genes in a subset of samples which were selected via mutation (i.e., you cannot see the frequency of mutations in specific genes that have a TP53 or KRAS mutation). I know this functionality would be nice to have, but I am not sure how to get that easily.

@pieterlukasse, to your last point: this looks like a bug, but let me think about it more.

**Comment by pieterlukasse** [comment]
@zhx828 are these issues solved in the new study view?

**Comment by zhx828** [comment]
@pieterlukasse yes! Thanks @awernibi and you brought up this issue.

**Comment by zhx828** [comment]
This issue should be solved in https://github.com/cBioPortal/cbioportal/pull/1405
